src/app/components/SEStatusLabel/SEStatusLabel.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+  VoidFunctionComponent,
+} from "react";
+import { ManagedResourceStatus } from "@openapi/generated";
+import { ResourceStatusLabel } from "@app/components/ResourceStatusLabel/ResourceStatusLabel";
+import {
+  ResourceStatus,
+  ResourceStatusDelayed,
+} from "@app/components/ResourceStatusLabel/types";
+
+import { useTranslation } from "react-i18next";
+import { differenceInMinutes } from "date-fns";
+import { usePolling } from "../../../hooks/usePolling/usePolling";
+import { CreationStatusOrder } from "@app/components/SEStatusLabel/SEStatusPopover/types";
+import SEStatusPopover from "@app/components/SEStatusLabel/SEStatusPopover/SEStatusPopover";
+
+export interface SEStatusLabelProps {
+  status: ManagedResourceStatus;
+  resourceType: "bridge" | "processor";
+  requestedAt: Date;
+}
+
+const SEStatusLabel: VoidFunctionComponent<SEStatusLabelProps> = (props) => {
+  const { status, resourceType, requestedAt } = props;
+  const { t } = useTranslation(["openbridgeTempDictionary"]);
+  const labelRef = useRef<HTMLButtonElement>(null);
+  const showPopover = CreationStatusOrder.indexOf(status) > -1;
+  const [alert, setAlert] = useState<ResourceStatusDelayed>();
+
+  const title =
+    resourceType === "bridge"
+      ? t("instance.creatingInstance")
+      : t("processor.creatingProcessor");
+
+  const resourceStatus = useMemo(() => {
+    switch (status) {
+      case ManagedResourceStatus.Accepted:
+      case ManagedResourceStatus.Preparing:
+      case ManagedResourceStatus.Provisioning:
+        return ResourceStatus.CREATING;
+      case ManagedResourceStatus.Deprovision:
+      case ManagedResourceStatus.Deleting:
+      case ManagedResourceStatus.Deleted:
+        return ResourceStatus.DELETING;
+      case ManagedResourceStatus.Failed:
+        return ResourceStatus.FAILED;
+      case ManagedResourceStatus.Ready:
+        return ResourceStatus.READY;
+    }
+  }, [status]);
+
+  const warningAfterMinutes = 5;
+  const errorAfterMinutes = 10;
```

Comment:
Setting relatively high time limits for the warnings for now.
I was also thinking, maybe this information could be returned by APIs? I mean the "delayed" condition, not the time limit.

Reply:
Can you explain a bit more?

Reply:
They are the two time limits from the request date after which the `This is taking longer than expected` warning/error message is displayed.
After 5 minutes -> warning (orange)
After 10 minutes -> error (red)

I was thinking that maybe we could have move this logic in the BE. Like having another optional property in the bridge/processor response like `delayed: "warning" | "error"` or something like that.

Did I answer your question?

Reply:
We are actually already polling the list of bridges/processors in the overview pages (every 10s atm). `SEStatusLabel` would then be refreshed accordingly.

Reply:
But we don't do it **everywhere** we use `SEStatusLabel` (and if we follow your proposal you would need to).

Reply:
Actually we have a slightly different situation.

If we consider the current implementation of this PR, when you create/update a processor you are kept in the processor detail page (not polled). If you remain there >5min, you will see the "taking longer" because the UI is displaying it based on the created date. And it could be not true. So I should fix it somehow here.

On the contrary, if the `delayed` info is a property on the processor response, after saving a processor you would just see the `Creating` message until you leave/refresh. It could be of course an outdated status if you stay for long, but that's a problem we already have today.

Reply:
LOL, yes I understand. By virtue of the not refreshing the label the initial state (when #1033 is merged) would be OK.

Reply:
Yes but I still have to tell `SEStatusLabel` to not update the "delayed" message over time in the processor detail page. Even after [#1033](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1033) it would not work properly as it is now :slightly_frowning_face: